lib/tools/broccoli/node-modules-dest-copy.ts

```diff
@@ -30,41 +30,43 @@ export class DestCopy implements IBroccoliPlugin {
   
   public rebuildChangedDirectories(changedDirectories: string[], platform: string): void {
 	_.each(changedDirectories, changedDirectoryAbsolutePath => {
-		let pathToPackageJson = path.join(changedDirectoryAbsolutePath, "package.json");
-		let packageJsonFiles = fs.existsSync(pathToPackageJson) ? [pathToPackageJson] : [];
-		let nodeModulesFolderPath = path.join(changedDirectoryAbsolutePath, "node_modules");
-		packageJsonFiles = packageJsonFiles.concat(this.enumeratePackageJsonFilesSync(nodeModulesFolderPath));
-			
-		_.each(packageJsonFiles, packageJsonFilePath => {
-			let fileContent = require(packageJsonFilePath);
-			let isPlugin = fileContent.nativescript;
-	
-			if(!this.devDependencies[fileContent.name]) { // Don't flatten dev dependencies
-		
-				let currentDependency = {
-					name: fileContent.name,
-					version: fileContent.version,
-					directory: path.dirname(packageJsonFilePath),
-					isPlugin: isPlugin
-				};
+		if(!this.devDependencies[path.basename(changedDirectoryAbsolutePath)]) {
```

Comment:
I would personally use `filter` here. This will make the code more readable and remove the code nesting with one level.